Assets/MixedRealityToolkit/Inspectors/Utilities/MixedRealitySearchUtility.cs

```diff
@@ -0,0 +1,225 @@
+        /// Searches all profiles and sub profiles for matching fields. Returns results in unranked order.
+        /// </summary>
+        /// <param name="profile">The configuration profile you're searching.</param>
+        /// <param name="config">Configuration for search.</param>
+        public static IEnumerable<ProfileSearchResult> SearchProfileFields(UnityEngine.Object profile, SearchConfig config)
+        {
+            if (string.IsNullOrEmpty(config.SearchFieldString))
+            {   // If the config is empty, bail early
+                yield break;
+            }
+
+            // Generate keywords if we haven't yet
+            if (config.Keywords == null)
+            {
+                config.Keywords = new HashSet<string>(config.SearchFieldString.Split(new string[] { " ", "," }, StringSplitOptions.RemoveEmptyEntries));
+                config.Keywords.RemoveWhere(s => s.Length < minSearchStringLength);
+            }
+
+            // The result that we will return, if not empty
+            ProfileSearchResult result = new ProfileSearchResult();
+
+            // Ignore these requirements if criteria is empty
+            config.RequireAllKeywords &= config.Keywords.Count > 0;
+
+            // Go through the profile's serialized fields
+            SerializedProperty iterator = new SerializedObject(profile).GetIterator();
+            bool hasNextProperty = iterator.Next(true);
+
+            while (hasNextProperty)
+            {
+                if (serializedPropertiesToIgnore.Contains(iterator.name))
+                {   // We don't care about this property
+                    hasNextProperty = iterator.Next(false);
+                    continue;
+                }
+
+                // Check to see if this is a profile - if it is, enter the profile and check its children
+                bool isProfileField = false;
+                if (iterator.propertyType == SerializedPropertyType.ObjectReference && iterator.objectReferenceValue != null)
+                {   // If the property is itself a profile, perform a recursive search
+                    Type referenceType = iterator.objectReferenceValue.GetType();
+                    if (typeof(BaseMixedRealityProfile).IsAssignableFrom(referenceType))
+                    {
+                        isProfileField = true;
+                        foreach (ProfileSearchResult subResult in SearchProfileFields(iterator.objectReferenceValue, config))
+                        {
+                            yield return subResult;
+                        }
+                    }
+                }
+
+                // Gather our top-level info
+                bool keywordReqsMet = !config.RequireAllKeywords;
+                int fieldMatchStrength = 0;
+
+                CheckFieldKeywords(config, iterator, ref fieldMatchStrength, ref keywordReqsMet);
+                if (keywordReqsMet)
+                {
+                    if (ProfileSearchResult.IsEmpty(result))
```

Comment:
the isempty here is a bit over-complicated. All you really need to track is the MaxFieldMatchStrength and a List<FieldSearchResult>. Why not declare these and then if your list is not empty at the end of the function, you create a ProfileSearchResult, populate, and return? 

This way you don't need to constantly check if isEmpty and can actually remove the function. You don't actually ever return an empty result so it's not utilized by the consumer of this utility either.